assets/js/modules/analytics-4/utils/data-mock.js

```diff
@@ -43,6 +43,7 @@ const ANALYTICS_4_METRIC_TYPES = {
 	conversions: 'TYPE_INTEGER',
 	screenPageViews: 'TYPE_INTEGER',
 	engagedSessions: 'TYPE_INTEGER',
+	engagementRate: 'TYPE_INTEGER',
```

Comment:
The type here is a float, see real example:

<img width="453" alt="image" src="https://user-images.githubusercontent.com/1621608/223229688-0c09ae78-25f8-4bba-982a-a2eb494311ac.png">

Looks like we'd need to add handling for this in `generateMetricValues` as well.

```suggestion
	engagementRate: 'TYPE_FLOAT',
```

Reply:
Thanks for spotting that! I've updated the widget, and the mock data generator accordingly.